redis/src/main/scala/zio/redis/RedisPubSubCommand.scala

```diff
@@ -0,0 +1,25 @@
+package zio.redis
+
+import zio.redis.Output.ArbitraryOutput
+import zio.schema.Schema
+import zio.schema.codec.BinaryCodec
+import zio.stream._
+import zio.{Chunk, IO, ZIO, ZLayer}
+
+private[redis] final case class RedisPubSubCommand(command: PubSubCommand, codec: BinaryCodec, executor: RedisPubSub) {
+  def run[R: Schema](): IO[RedisError, Chunk[Stream[RedisError, R]]] = {
+    val codecLayer = ZLayer.succeed(codec)
+    executor
+      .execute(command)
```

Comment:
Can you do the encoding here and avoid passing the codec as a layer?

Reply:
it can be removed from `RedisPubSub` interface but `SignleNodeRedisPubSub` needs codec to managing sub/unsub state and re-subscribe for failover so I moved codec into `SignleNodeRedisPubSub`'s field